storage/cache/subtree_cache.go

```diff
@@ -235,25 +231,22 @@ func (s *SubtreeCache) SetNodes(nodes []tree.Node, getSubtrees GetSubtreesFunc)
 	return nil
 }
 
-// Flush causes the cache to write all dirty Subtrees back to storage.
-func (s *SubtreeCache) Flush(ctx context.Context, setSubtrees SetSubtreesFunc) error {
-	treesToWrite := make([]*storagepb.SubtreeProto, 0)
+// Dirty returns all dirty Subtrees that need to be written back to storage.
+func (s *SubtreeCache) Dirty() ([]*storagepb.SubtreeProto, error) {
```

Comment:
Naming suggestions welcome. How about `Prepare`?

Reply:
Some more suggestions:
`ModifiedSubtrees` / `UpdatedSubtrees` / `NewSubtrees`
`ToBeWritten`

Reply:
Renamed to `UpdatedTiles`.